crates/sui-core/src/epoch/committee_store.rs

```diff
@@ -54,11 +53,10 @@ impl CommitteeStore {
     }
 
     pub fn insert_new_committee(&self, new_committee: &Committee) -> SuiResult {
-        let latest_committee = self.get_latest_committee();
-        fp_ensure!(
-            latest_committee.epoch + 1 == new_committee.epoch,
-            SuiError::from("Unexpected new epoch number")
-        );
+        if let Some(old_committee) = self.get_committee(&new_committee.epoch)? {
+            // If somehow we already have this committee in the store, they must be the same.
+            assert_eq!(&old_committee, new_committee);
```

Comment:
Should we return from the function after this assert?

Reply:
It's an assert, which would panic?

Reply:
I mean do we still need the insert below if the existing committee in the db is same as what we are going to insert again.

Reply:
When this happens, we probably have a fork. So it probably doesn't matte which committee we keep for this epoch.

Reply:
I think @sadhansood is referring to a fact that if we did have a committee and it does match current value, we could have a separate branch for this case and just skip call to `insert`(which will save some serialization/io).